proto/config/storage.proto

```diff
@@ -171,8 +171,9 @@ enum VolumeAccessProtocols {
 
 enum VolumeContentOriginType {
   VCOT_UNKNOWN = 0;
-  VCOT_BLANK = 1;     // Create an empty directory/disk with specified size
-  VCOT_DOWNLOAD = 2;  // Create from downloadContentTreeID
+  VCOT_BLANK = 1;       // Create an empty directory/disk with specified size
+  VCOT_DOWNLOAD = 2;    // Create from downloadContentTreeID
+  VCOT_REPLICATED = 3;  // Use replicated volume from other nodes
```

Comment:
but if it's 'replicated' flag set, do we have no need for 'download' or 'blank' flags anymore?

Reply:
Yes for replicated volumes. But there will be non-replicated volumes or this node could be master of a replicated volume so that will still have download or blank

Reply:
But do we want to support blank replicated volumes e.g., for a data volume which is initially empty?

Or is the intent that one of the nodes receives the config with DOWNLOAD or BLANK, and the other nodes in the cluster receives the config with thy type set to REPLICATED?

Reply:
> Or is the intent that one of the nodes receives the config with DOWNLOAD or BLANK, and the other nodes in the cluster receives the config with thy type set to REPLICATED?

Yes this is the intent. A node creating the volume will get BLANK or DOWNLOADED, other nodes will get REPLICATED type in the config

Reply:
But that approach does not enable an autonomous cluster during the creation (and I suspect it would be complex with race condition to have the controller assist with recovery should the creating node fail and either come back alive or not).

So I think the right approach here as well as elsewhere in the API is to tell all the nodes in the cluster the intended state (which in this case is that a volume should exist across the cluster) and if need he add hits and suggestions from the controller (such as the device uuid if the suggested creator - sent to all the nodes).

With such an API approach the initial implementation can rely solely on the suggested creator, but later we can make the cluster elect to ignore that should that node die. Thus we can improve robustness and autonomy of the cluster without neither adding to the API nor making the controller do something to help in recovery.

My intention was to have all aspects of the cluster API additions be done following that approach. I don't know if we did that for the previous additions but it might make sense to take a look.